mlir/lib/Dialect/Linalg/Transforms/ElementwiseOpFusion.cpp

```diff
@@ -216,16 +217,17 @@ bool mlir::linalg::areElementwiseOpsFusable(OpOperand *fusedOperand) {
 /// Generate the region of the fused tensor operation. The region of the fused
 /// op must be empty.
 static void generateFusedElementwiseOpRegion(
-    RewriterBase &rewriter, GenericOp fusedOp,
+    RewriterBase &rewriter, LinalgOp fusedOp,
     AffineMap consumerToProducerLoopsMap, OpOperand *fusedOperand,
     unsigned nloops, llvm::SmallDenseSet<int> &preservedProducerResults) {
-  auto producer = cast<GenericOp>(fusedOperand->get().getDefiningOp());
-  auto consumer = cast<GenericOp>(fusedOperand->getOwner());
+  auto producer = cast<LinalgOp>(fusedOperand->get().getDefiningOp());
+  auto consumer = cast<LinalgOp>(fusedOperand->getOwner());
   // Build the region of the fused op.
+
   Block &producerBlock = producer->getRegion(0).front();
   Block &consumerBlock = consumer->getRegion(0).front();
```

Comment:
You cannot do this on an interface. There is no gurantee that the op implementing this interface h: https://github.co region 0 as the region of th interface. You can use the interface methods to get this: https://github.com/llvm/llvm-project/blob/cdf52a1325e6329cef267ca93886481ddc5d554d/mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td#L491

Reply:
done